### PR TITLE
[Feature][plugin-filter-date] Date filter supports Timestamp.

### LINKED ISSUE #852: [Feature][plugin-filter-date] Date filter supports Timestamp
### Search before asking

- [X] I had searched in the [feature](https://github.com/InterestingLab/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

The filter of Date supports convert from `Timestamp`

### Usage Scenario

date {
    source_field = "timestamp_field"
    target_field = "date"
    source_time_format = "TIMESTAMP"
    target_time_format = "yyyy/MM/dd"
}

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by dongzl**
## Purpose of this pull request

Fixes #852 .

### CODE REVIEW

**Review by RickyHuo — CHANGES_REQUESTED** [review]
`TimestampParser` seems invalid. You can test the demo 

```
spark {
  spark.app.name = "seatunnel"
  spark.executor.instances = 2
  spark.executor.cores = 1
  spark.executor.memory = "1g"
}

input {
  # This is a example input plugin **only for test and demonstrate the feature input plugin**
  fake {
    result_table_name = "my_dataset"
  }

}

filter {

  sql {
    sql = "select timestamp'2019-08-12 01:00:00.123456' as t, * from my_dataset"
  }

  date {
      source_field = "t"
      target_field = "format_d"
      source_time_format = "TIMESTAMP"
      target_time_format = "yyyy/MM/dd hh:mm:ss."
  }
}

output {
  stdout {

  }

  stdout {
    format = "schema"
  }
}
```

File: waterdrop-core/src/main/scala/io/github/interestinglab/waterdrop/utils/TimestampParser.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import scala.util.control.NonFatal
+
+class TimestampParser(targetTimeFormat: String) extends DateParser {
+
+  val targetFormat = targetTimeFormat
+
+  def parse(input: String): (Boolean, String) = {
+
+    try {
+      val timeMillis = input.toLong
```
Comment (RickyHuo) [inline waterdrop-core/src/main/scala/io/github/interestinglab/waterdrop/utils/TimestampParser.scala:35]: The **input** can be converted to long

File: waterdrop-core/src/main/scala/io/github/interestinglab/waterdrop/utils/TimestampParser.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import scala.util.control.NonFatal
+
+class TimestampParser(targetTimeFormat: String) extends DateParser {
+
+  val targetFormat = targetTimeFormat
+
+  def parse(input: String): (Boolean, String) = {
+
+    try {
+      val inputDateFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss.SSSSSS")
```
Comment (RickyHuo) [inline waterdrop-core/src/main/scala/io/github/interestinglab/waterdrop/utils/TimestampParser.scala:34]: I think `yyyy-MM-dd hh:mm:ss.SSS` is enough. I'm sorry i gave the error timestamp demo

**Review by RickyHuo — CHANGES_REQUESTED** [review]

File: waterdrop-core/src/main/scala/io/github/interestinglab/waterdrop/utils/TimestampParser.scala (hunk 2 above)
Comment (dongzl) [reply waterdrop-core/src/main/scala/io/github/interestinglab/waterdrop/utils/TimestampParser.scala]: OK, I fix it. Thanks.

**Review by RickyHuo — APPROVED** [review]
LGTM

**Pull request merged**